### DISCUSSION TITLE
Issue while using the puppeteer library.

### DISCUSSION THREAD

**Discussion by aadityam16** [supabase/supabase#21043] [Questions]
This is my web-scraper edge function (index.ts):

import puppeteer from "https://deno.land/x/puppeteer@16.2.0/mod.ts";

Deno.serve(async (req) => {
  try {
    const { url } = await req.json();
    
    const scrape = async (url: string) => {
      const browser = await puppeteer.launch();
      const page = await browser.newPage();
  
      await page.goto(url);
  
      const title = await page.title();
  
      const emails = await page.evaluate(() => {
        const emailRegex = /\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/g;
        const bodyText = document.body.innerText;
        return bodyText.match(emailRegex);
      });
  
      const divData = await page.evaluate(() => {
        const divs = Array.from(document.querySelectorAll("div"));
  
        const extractDataFromDiv = (div: HTMLDivElement) => {
          const divContent = div.innerText;
  
          const splitContent = divContent
            .split("\n")
            .map((content) => content.trim())
            .filter((content) => content !== "" && content.trim() !== "");
  
          return splitContent;
        };
  
        return divs.map(extractDataFromDiv);
      });
  
      await browser.close();
  
      return {
        title,
        emails,
        content: divData,
      };
    };
  
    const data = await scrape(url);
  
    return new Response(
      JSON.stringify(data),
      { headers: { "Content-Type": "application/json" } },
    );
  } catch (error) {
    return new Response(`Error: ${error}`, { status: 500 });
  }
});

I deployed this function to my supabase,
here is the error which is showing in postman:

<img width="642" alt="image" src="https://github.com/supabase/supabase/assets/95746844/e930264f-c5df-45ca-89df-f5600ad69084">

Someone please help

**Comment by mwoss** [comment]
As far as I know, edge functions don't have access to the file system for security reasons, thus all file-system-related functionalities are turned off. I think I saw a similar issue somewhere in the past, I'll try to find it.

**Reply by mwoss** [reply]
Yup, take a look at this one https://github.com/supabase/edge-runtime/issues/120
